lib/KTable/index.vue

```diff
@@ -294,156 +294,147 @@
-                } else {
-                  // Allow default behavior when reaching the last cell
-                  return;
-                }
-              } else {
-                if (colIndex > 0) {
-                  nextColIndex = colIndex - 1;
-                } else if (rowIndex > 0) {
-                  nextColIndex = totalCols - 1;
-                  nextRowIndex = rowIndex - 1;
-                } else {
-                  // Allow default behavior when reaching the first cell
-                  return;
-                }
-              }
+              return; // Allow default Tab behavior when reaching the end
             }
-
-            break;
+            const nextCell = this.getCell(nextRowIndex, nextColIndex);
+            const nextFocusableElements = this.getFocusableElements(nextCell);
+            const nextCellAndFocusableElements = [nextCell, ...nextFocusableElements];
+            nextCellAndFocusableElements[0].focus();
+            this.updateFocusState(nextRowIndex, nextColIndex);
+            event.preventDefault();
           }
-
-          default:
+        } else {
+          // Shift+Tab key navigation
+          if (focusedIndex > 0) {
+            // Move to the previous focusable element within the cell
+            cellAndFocusableElements[focusedIndex - 1].focus();
+            event.preventDefault();
             return;
+          } else {
+            // Move to the last focusable element of the previous cell
+            if (colIndex > 0) {
+              nextColIndex = colIndex - 1;
+            } else if (rowIndex > 0) {
+              nextColIndex = totalCols - 1;
+              nextRowIndex = rowIndex - 1;
+            } else {
+              return; // Allow default Shift+Tab behavior when at the beginning
+            }
+            const prevCell = this.getCell(nextRowIndex, nextColIndex);
+            const prevFocusableElements = this.getFocusableElements(prevCell);
+            const prevCellAndFocusableElements = [prevCell, ...prevFocusableElements];
+            prevCellAndFocusableElements[prevCellAndFocusableElements.length - 1].focus();
+            this.updateFocusState(nextRowIndex, nextColIndex, false);
+            event.preventDefault();
+          }
         }
-
-        this.focusCell(nextRowIndex, nextColIndex);
+      },
+      updateFocusState(nextRowIndex, nextColIndex, shouldFocusCell = true) {
         this.focusedRowIndex = nextRowIndex === -1 ? null : nextRowIndex;
         this.focusedColIndex = nextColIndex;
-
         this.highlightHeader(nextColIndex);
+        // Focus the cell only if it is necessary
+        if (shouldFocusCell) {
+          this.focusCell(nextRowIndex, nextColIndex);
+        }
+      },
 
-        event.preventDefault();
+      getFocusableElements(cell) {
+        if (!cell) return [];
+        const selectors = 'button, a, input, select, textarea';
+        return Array.from(cell.querySelectorAll(selectors));
```

Comment:
About the use of `querySelectorAll` here, I was actually having a little lag in the navigation behaviour when moving through the focusable elements - this reminded me of an earlier code review comment [here](https://github.com/learningequality/kolibri-design-system/pull/727#discussion_r1720879608). We were using querySelectorAll earlier as well, but it caused performance issues, so we shifted to `getElementByTagname` and it had a very positive impact on the same. Can you update this code to make use of the previous getElementbyTagname logic? or maybe if there's any other conclusion you derive from that conversation about performance improvement- it is more than welcome. Thanks!

Reply:
@BabyElias, Thanks for noticing. I've changed it to `getElementsByTagName`.